src/app/components/ConsensusTransactionMethod/index.tsx

```diff
@@ -75,19 +99,93 @@ export const MethodIcon: FC<MethodIconProps> = ({
       >
         {cloneElement(icon, { sx: { fontSize: Math.ceil(size * iconRatio) } })}
       </Box>
-      {label && <Typography sx={{ textTransform: 'capitalize' }}>{label}</Typography>}
+      {label && (
+        <Typography
+          sx={{
+            fontWeight: 'inherit',
+            textTransform: 'capitalize',
+            order: reverseLabel ? -1 : 0,
+            ...(truncate
+              ? {
+                  minWidth: '100px',
+                  maxWidth: '1px',
+                  whiteSpace: 'nowrap',
+                  overflow: 'hidden',
+                  textOverflow: 'ellipsis',
+                }
+              : {}),
+          }}
+        >
+          {label}
+        </Typography>
+      )}
     </Box>
   )
 }
 
-const getConsensusTransactionMethod = (t: TFunction, method: ConsensusTxMethod | undefined) => {
+const MethodIconWithTruncatedLabel: FC<MethodIconProps> = props => {
+  const elementRef = useRef<HTMLDivElement>(null)
+  const [truncate, setTruncate] = useState(false)
+  const [baseLabelWidth, setBaseLabelWidth] = useState(0)
+  const applyTruncate = useCallback(() => {
+    if (elementRef.current) {
+      const elementWidth = elementRef.current.offsetWidth
+      const parentWidth = (elementRef.current.parentNode as HTMLElement)?.offsetWidth
+      setTruncate(elementWidth === parentWidth || baseLabelWidth > parentWidth)
+    }
+  }, [elementRef, baseLabelWidth])
+
+  useLayoutEffect(() => {
+    if (elementRef.current) {
+      setBaseLabelWidth(elementRef.current.offsetWidth)
+      applyTruncate()
+    }
+
+    const handleResize = () => {
+      applyTruncate()
+    }
+
+    window.addEventListener('resize', handleResize)
+
+    return () => {
+      window.removeEventListener('resize', handleResize)
+    }
+  }, [applyTruncate, elementRef])
```

Comment:
This is not needed for our case

Reply:
Sure, but that's just by luck because so far it's only used in table rows with key=transaction.hash.

If we enabled truncating on RuntimeTransactionMethod in RuntimeTransactionDetailPage then open transaction 286f746049606e6cb743d381ee1ca4c810db44fcda6ba6fbb1d7a1c84a53c1a9 then search for 0x5f7de27c3d47551fc2aea0934b747cd76e5358ab4cbfced2015f67cd194f82f8: the element is reused and state is wrong